SmartBadge/lib/ext/bt/btcomms.py

```diff
@@ -10,9 +10,7 @@ class _BlueComm(object):
 
     def __init__(self):
         self.ble = bluetooth.BLE()
-        # self.display_name = 'Badge-'+''.join(['{:02x}'.format(b) for b in machine.unique_id()])
-        self.display_name = ''.join(['{:02x}'.format(b) for b in machine.unique_id()])
-        self.display_name = self.display_name[-4:]
+        self.display_name = 'SB'
```

Comment:
Do all devices have this name then? How do we uniquely identify them?

Reply:
The name is a just user-friendly identifier. I have decided to implement the system using Bluetooth UUIDs, which specify the type of service being operated. Only devices matching the UART service will be provided as options in the list returned from `detected_devices()`. Users can then pick a device to connect to based on the MAC addresses of the devices (which are unique). The MAC addr. is sent with the advertising payload, so it is available to the master. The slave can check its own MAC, so both parties can use this mechanism to determine who to connect to.

---

Side note: the name originally contained the MAC addr., but this made the advertising payload (which contains name and service UUID) too long (max. 29 bytes), so an alternative system was needed.